app/models/os_major_version.rb

```diff
@@ -19,22 +45,22 @@ class OsMajorVersion < ApplicationRecord
                         foreign_key: 'ref_id', primary_key: 'ref_id',
                         inverse_of: false, dependent: :restrict_with_exception
 
+  has_many :profiles, lambda {
+    supported_profiles = canonical.where(upstream: false)
+                                  .joins(:benchmark)
+                                  .select(PROFILE_LAST_ID, PROFILE_BM_VERSIONS, PROFILE_BM_REF_ID)
+                                  .group(:ref_id, 'benchmarks.ref_id')
+
+    canonical.where(upstream: false)
+             .joins("INNER JOIN (#{supported_profiles.to_sql}) t ON t.id = profiles.id AND t.bm = benchmarks.ref_id")
```

Comment:
What do we need the profile `id` for? Can we match on profile's ref_id instead? Double-check the [example](https://github.com/RedHatInsights/compliance-backend/pull/1114#discussion_r791681248).

Reply:
Because otherwise it returns with **one** profile matching the `ref_id` and benchmark, but not the **latest supported one**, which is desirable IMO. I don't want the outer query to be affected with sorting/grouping as there will be sorting/grouping requested by GQL (look at the UX mockup).

Reply:
Also this will eventually allow us to eliminate the N+1 if I set up the `supported_profiles` as a reflection. Still working on it, not part of this PR...